### PR TITLE
[LG-2424, LG-2286, LG-1783] Update Copyable shadow. Forward Ref to InlineCode

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Shows copyable shadow only when text overflows.

Adds `forwardRef` to `InlineCode`.
Defaults `InlineCode` display property to `inline`

🎟 _Jira ticket:_ [Copyable shadow](https://jira.mongodb.org/browse/LG-2286)
🎟 _Jira ticket:_ [InlineCode forwardRef](https://jira.mongodb.org/browse/LG-1783)
🎟 _Jira ticket:_ [InlineCode display](https://jira.mongodb.org/browse/LG-2424)

Storybook Demo: https://thesonofthomp.github.io/leafygreen-ui/?path=/story/components-copyable--basic
Screen Shot 2022-11-02 at 11 42 57 AM
Screen Shot 2022-11-02 at 11 42 49 AM

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/copyable/src/Copyable.tsx [hunk 1]
```diff
@@ -227,37 +68,11 @@ export default function Copyable({
...
-        justify={Justify.Middle}
-        trigger={trigger}
-        triggerEvent={TriggerEvent.Click}
-        usePortal={shouldTooltipUsePortal}
-      >
-        Copied!
-      </Tooltip>
-    );
-  }
+  const isOverflowed = codeRef.current
```
Comment (shaneeza) [inline packages/copyable/src/Copyable.tsx:71]: 👍🏽

File: .changeset/light-taxis-sleep.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/typography': patch
```
Comment (shaneeza) [inline .changeset/light-taxis-sleep.md:2]: In the ticket you mentioned that this will be a major change, why the switch to patch?

**Review by shaneeza — APPROVED** [review]
Looks good, left a minor comment!

File: .changeset/light-taxis-sleep.md (hunk 2 above)
Comment (TheSonOfThomp) [reply .changeset/light-taxis-sleep.md:2]: Ah, missed that. Will fix 👍

Comment (TheSonOfThomp) [reply .changeset/light-taxis-sleep.md:2]: It should always have been `inline`, so this is just a bug fix. Not sure why I wrote that in the ticket

**Pull request merged**